File: problem015/solution.php
<?php

require __dir__ . "/../src/helpers.php";

/*
 * The number of lattice paths from (0,0)} to (n,k) is equal to the binomial
 * coefficient (n + k choose k). We can use the multiplicative formula to
 * calculate the coefficient.
 */

$n = 20;
$k = 20;

for ($paths = 1, $i = 1; $i <= $k; $i++) {
    $paths *= (($n + $k) + 1 - $i) / $i;
}

solution($paths);

File: problem015/solution.py
from src import helpers

# The number of lattice paths from (0,0)} to (n,k) is equal to the binomial
# coefficient (n + k choose k). We can use the multiplicative formula to
# calculate the coefficient.

n = 20
k = 20
paths = 1

for i in range(1, k + 1):
    paths *= ((n + k) + 1 - i) / i

helpers.solution(int(paths))
